thundra/plugins/metric/samplers/time_aware_metric_sampler.py

```diff
@@ -0,0 +1,21 @@
+import time
+from thundra import constants, config
+
+class TimeAwareMetricSampler:
+
+    def __init__(self, time_freq=None):
+        freq_from_env = config.time_aware_metric_freq()
+        if freq_from_env > 0:
+            self.time_freq = freq_from_env
+        elif time_freq is not None:
+            self.time_freq = time_freq
+        else:
+            self.time_freq = constants.DEFAULT_METRIC_SAMPLING_TIME_FREQ
+        self._latest_time = 0
+
+    def is_sampled(self):
+        current_time = 1000 * time.time()
```

Comment:
not important but it would be good if we could have lock around this as currently the logic is not thread-safe